clang/lib/DPCT/DPCT.cpp

```diff
@@ -542,6 +543,156 @@ void parseFormatStyle() {
   DpctGlobalInfo::setCodeFormatStyle(Style);
 }
 
+static std::string getCmakeBuildPathFromInRoot(StringRef InRoot,
+                                               StringRef OutRoot) {
+  std::error_code EC;
+
+  std::string CmakeBuildDirectory;
+  for (fs::recursive_directory_iterator Iter(Twine(InRoot), EC), End;
+       Iter != End; Iter.increment(EC)) {
+    if ((bool)EC) {
+      std::string ErrMsg = "[ERROR] Access : " + std::string(InRoot.str()) +
+                           " fail: " + EC.message() + "\n";
+      PrintMsg(ErrMsg);
+    }
+
+    auto FilePath = Iter->path();
+
+    // Skip output directory if it is in the in-root directory.
+    if (isChildOrSamePath(OutRoot.str(), FilePath))
+      continue;
+
+    bool IsHidden = false;
+    for (path::const_iterator PI = path::begin(FilePath),
+                              PE = path::end(FilePath);
+         PI != PE; ++PI) {
+      StringRef Comp = *PI;
+      if (Comp.startswith(".")) {
+        IsHidden = true;
+        break;
+      }
+    }
+    // Skip hidden folder or file whose name begins with ".".
+    if (IsHidden) {
+      continue;
+    }
+
+    if (Iter->type() == fs::file_type::directory_file) {
+      const auto Path = Iter->path();
+      SmallString<512> OutDirectory = llvm::StringRef(Path);
+      if (fs::exists(OutDirectory + "/CMakeFiles") &&
```

Comment:
I suggest to use path::append() instead of using "+"

Reply:
okay